manifest: Update hal_nxp to add Wi-Fi driver

MaochenWang1 opened the pull request:
Update hal_nxp to add Wi-Fi driver code.

MaochenWang1 committed:
manifest: Update hal_nxp to add Wi-Fi driver

Update hal_nxp to add Wi-Fi driver code.

Signed-off-by: Maochen Wang <maochen.wang@nxp.com>

bjarki-andreasen commented:
looks good, ping when its a sha :)

danieldegrasse committed:
drivers: wifi: nxp: add CONFIG_NXP_WIFI_BUILD_ONLY_MODE for CI testing

Add CONFIG_NXP_WIFI_BUILD_ONLY_MODE, which allows the NXP WiFi driver to
be built without a binary blob. The built application will not be
functional, but this enables continuous integration to run upstream.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
modules: hostap: remove ap_bandwidth API

ap_bandwidth API implementation is not complete. Remove the structure
definition and implementation from the NXP WiFi driver to fix a build
error.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
modules: hostap: remove set_btwt API

set_btwt API implementation is not complete. Remove the structure
definition and implementation from the NXP WiFi driver to fix a build
error.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
drivers: wifi: nxp: remove non-existent TWT fields

Remove references to TWT fields that did not exist, to resolve a build
error.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
samples: net: wifi: add support for building NXP WiFi

Add support for building NXP WiFi. Since the driver typically requires
binary blobs, enable CONFIG_NXP_WIFI_BUILD_ONLY_MODE for this testcase.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
drivers: wifi: nxp: remove unused nxp_wifi_uap_status function

Remove unused function `nxp_wifi_uap_status`.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse commented:
Updated this PR with the contents of #80082, as the WiFi driver was previously not building in CI (and did not build without the changes now applied to this PR)

krish2718 commented on modules/hostap/src/supp_main.h:
```diff
@@ -15,7 +15,7 @@
 	!defined(CONFIG_EAP_SAKE) && !defined(CONFIG_EAP_GPSK) && \
 	!defined(CONFIG_EAP_PWD) && !defined(CONFIG_EAP_EKE) && \
-	!defined(CONFIG_EAP_IKEV2 && !defined(CONFIG_EAP_GTC)
+	!defined(CONFIG_EAP_IKEV2) && !defined(CONFIG_EAP_GTC)
```
Also fixed by https://github.com/zephyrproject-rtos/zephyr/pull/79974/commits/de54470bac24e8f6d201105a12ff867fd38d383f

krish2718 replied:
@jukkar how do you want deal with this? Merge this and abandon that PR or wait for it to be merged and rebase this?

dleach02 replied:
I think you should let the other PR still move forward. @jukkar PR is needed to add the test case going forward.

krish2718 commented on modules/hostap/src/supp_main.c:
```diff
@@ -98,7 +98,6 @@ static const struct wifi_mgmt_ops mgmt_ap_ops = {
 	.ap_enable = supplicant_ap_enable,
 	.ap_disable = supplicant_ap_disable,
 	.ap_sta_disconnect = supplicant_ap_sta_disconnect,
-	.ap_bandwidth = supplicant_ap_bandwidth,
```
please make a separate commit for hostap, and I guess the missing definition `supplicant_ap_bandwidth` is what is incomplete? and this is an accidental change?

krish2718 commented on modules/hostap/src/supp_main.c:
```diff
@@ -94,7 +94,6 @@ DEFINE_WIFI_NM_INSTANCE(wifi_supplicant, &mgmt_ops);
 
 #ifdef CONFIG_WIFI_NM_HOSTAPD_AP
 static const struct wifi_mgmt_ops mgmt_ap_ops = {
-	.set_btwt = supplicant_set_btwt,
```
same as AP bandwidth, an accidental change now being reverted?

MaochenWang1 replied:
this is caused by wrong upstream sequence from downstream

krish2718 commented on samples/net/wifi/boards/rd_rw612_bga.conf:
```diff
@@ -75,33 +75,6 @@ CONFIG_NET_SOCKETS_SERVICE_THREAD_PRIO=3
 CONFIG_NET_TC_SKIP_FOR_HIGH_PRIO=y
 CONFIG_NET_CONTEXT_PRIORITY=y
 
-CONFIG_WIFI_NM_WPA_SUPPLICANT=y
```
but without this will Wi-Fi work on RW612?

MaochenWang1 replied:
remove this commit

krish2718 commented on west.yml:
```diff
@@ -198,7 +198,7 @@ manifest:
         - hal
     - name: hal_nxp
-      revision: 4597b16cfedf5553cb155151e65eb994d5d0ef25
+      revision: pull/454/head
```
is this the right PR? I don't see any Wi-Fi driver source files added to the HAL, just the CI flag which is linked to another manifest PR?

MaochenWang1 replied:
wifi driver already merged into hal_nxp, now this links to another PR to fix the build issue, which is caused by one hostap PR still under review.

dleach02 replied:
@krish2718, we were in the process of merging things when we found that there was a build problem. So we added the appropriate fix on this side and a change so that CI will actually build this platform. We used a path that Nordic did for platforms that require a blob to do a "build only" kconfig so CI would at least catch the error.

krish2718 replied:
Sure, I was confused by the title that `Wi-Fi driver supprot added to HAL` but in that PR didn't see those changes.

MaochenWang1 committed:
hostap: fix build error when WPS enabled

Fix build error of multiple definition of `dh5_init' in
crypto_mbedtls_alt.c when WPS enabled.

Signed-off-by: Maochen Wang <maochen.wang@nxp.com>

MaochenWang1 committed:
samples: wifi: fix build warning of DIV_ROUND_UP redefined

Remove CONFIG_MBEDTLS_HAVE_TIME_DATE, which will cause that mbedtls
include Zephyr time.h and leads to 'DIV_ROUND_UP' redefined warning.

Signed-off-by: Maochen Wang <maochen.wang@nxp.com>

danieldegrasse committed:
net: l2: wifi: add explicit dependency on "app" target for pem headers

Generated PEM headers need to be created before we build the WiFi shell
when WIFI_NM_WPA_SUPPLICANT_CRYPTO_ENTERPRISE is enabled, so add an
explicit dependency on the app target to be certain these headers are
created

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

MaochenWang1 commented:
Hi @jukkar  @krish2718  please review again

dleach02 commented:
@bjarki-andreasen it is a sha

krish2718 commented on drivers/wifi/nxp/Kconfig.nxp:
```diff
@@ -20,6 +20,13 @@ module = WIFI_NXP
 	default 50
 
+config NXP_WIFI_BUILD_ONLY_MODE
+	bool "Build only mode (do not link wireless binary blob)"
```
```suggestion
	bool "Build only mode (do not link wireless firmware blob)"
```

danieldegrasse replied:
Fixed, thanks

dleach02 commented:
CI failure is with a new test that merged between last night and resubmission of the PR with a SHA. Working to fix it now :-(

MaochenWang1 commented:
Hi @krish2718  @bjarki-andreasen pls review again
